zuul-core/src/main/java/com/netflix/zuul/util/HttpUtils.java

```diff
@@ -104,16 +104,21 @@ public static boolean acceptsGzip(Headers headers) {
      * @return - clean header string
      */
     public static String stripMaliciousHeaderChars(@Nullable String input) {
+        StringBuilder strippedInput = new StringBuilder();
+
         if (input == null) {
             return null;
         }
-        // TODO(carl-mastrangelo): implement this more efficiently.
-        for (char c : MALICIOUS_HEADER_CHARS) {
-            if (input.indexOf(c) != -1) {
-                input = input.replace(Character.toString(c), "");
+
+        for(char inputChar : input.toCharArray()){
```

Comment:
This would result in the input string being rebuilt each call to `stripMaliciousHeaderChars`.  The previous implementation had a fast path for no invalid characters.